lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -64,8 +64,15 @@ def disconnect_with_host_pooling!
 
     private
 
+    attr_accessor :_no_switch
+
     def _switch_connection
-      if _host_pool_current_database && ((_host_pool_current_database != @_cached_current_database) || @connection.object_id != @_cached_connection_object_id)
+      cached = defined?(@_cached_current_database) && @_cached_current_database
```

Comment:
do I just prepend to all the constructors? If this was just a boring old class that would be easy but it is used here: https://github.com/zendesk/active_record_host_pool/blob/roman/warnings/lib/active_record_host_pool/connection_adapter_mixin.rb#L137

Would appreciate some advice how to do that without an awful monkey patch.